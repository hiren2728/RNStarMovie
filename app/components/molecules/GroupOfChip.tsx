import React from "react"
import {
    View
} from "react-native"

// Component
import Chip from "../atom/Chip";

type Props = {
    chips: string[]
}

const GroupOfChip = ({chips} : Props) => {
    return (
        <View className={`flex-row flex-wrap items-baseline`}>
            {
                chips.map((item, index) => <Chip title={item} key={`chip_${index}`}/>)
            }
        </View>
    )
};

export default React.memo(GroupOfChip)
