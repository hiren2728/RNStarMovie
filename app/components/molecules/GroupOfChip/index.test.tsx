import React from "react";
import renderer from "react-test-renderer"
import GroupOfChip from "./index";


test('GroupOfChips renders correctly', () => {
    const tree = renderer.create(<GroupOfChip chips={["Hiren", "Work"]}/>).toJSON();
    expect(tree).toMatchSnapshot();
});
