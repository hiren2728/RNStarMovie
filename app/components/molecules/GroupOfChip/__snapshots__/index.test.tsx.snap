// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupOfChips renders correctly 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
      },
      {
        "flexWrap": "wrap",
      },
      {
        "alignItems": "baseline",
      },
    ]
  }
>
  <View
    style={
      [
        {
          "height": 28,
        },
        {
          "paddingLeft": 10,
          "paddingRight": 10,
        },
        {
          "justifyContent": "center",
        },
        {
          "marginBottom": 4,
          "marginLeft": 4,
          "marginRight": 4,
          "marginTop": 4,
        },
        {
          "borderBottomWidth": 1,
          "borderLeftWidth": 1,
          "borderRightWidth": 1,
          "borderTopWidth": 1,
        },
        {
          "borderBottomLeftRadius": 9999,
          "borderBottomRightRadius": 9999,
          "borderTopLeftRadius": 9999,
          "borderTopRightRadius": 9999,
        },
        {
          "borderBottomColor": "#A29CF5",
          "borderLeftColor": "#A29CF5",
          "borderRightColor": "#A29CF5",
          "borderTopColor": "#A29CF5",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#5C5A5B",
          },
          {
            "fontFamily": "RobotoMono-Bold",
          },
          [
            [
              {
                "fontSize": 12,
                "lineHeight": 16,
              },
              {
                "color": "#A29CF5",
              },
            ],
          ],
        ]
      }
    >
      Hiren
    </Text>
  </View>
  <View
    style={
      [
        {
          "height": 28,
        },
        {
          "paddingLeft": 10,
          "paddingRight": 10,
        },
        {
          "justifyContent": "center",
        },
        {
          "marginBottom": 4,
          "marginLeft": 4,
          "marginRight": 4,
          "marginTop": 4,
        },
        {
          "borderBottomWidth": 1,
          "borderLeftWidth": 1,
          "borderRightWidth": 1,
          "borderTopWidth": 1,
        },
        {
          "borderBottomLeftRadius": 9999,
          "borderBottomRightRadius": 9999,
          "borderTopLeftRadius": 9999,
          "borderTopRightRadius": 9999,
        },
        {
          "borderBottomColor": "#A29CF5",
          "borderLeftColor": "#A29CF5",
          "borderRightColor": "#A29CF5",
          "borderTopColor": "#A29CF5",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#5C5A5B",
          },
          {
            "fontFamily": "RobotoMono-Bold",
          },
          [
            [
              {
                "fontSize": 12,
                "lineHeight": 16,
              },
              {
                "color": "#A29CF5",
              },
            ],
          ],
        ]
      }
    >
      Work
    </Text>
  </View>
</View>
`;
